bigquery/src/main/java/org/apache/zeppelin/bigquery/bigQueryInterpreter.java

```diff
@@ -0,0 +1,340 @@
+  private static final char NEWLINE = '\n';
+  private static final char TAB = '\t';
+  private static Bigquery service = null;
+  //Mutex created to create the singleton in thread-safe fashion.
+  private static Object serviceLock = new Object();
+
+  static final String PROJECT_ID = "zeppelin.bigquery.project_id";
+  static final String WAIT_TIME = "zeppelin.bigquery.wait_time";
+  static final String MAX_ROWS = "zeppelin.bigquery.max_no_of_rows";
+
+  private static String jobId = null;
+  private static String projectId = null;
+
+  private static final List NO_COMPLETION = new ArrayList<>();
+  private Exception exceptionOnConnect;
+
+  private static final Function<CharSequence, String> sequenceToStringTransformer =
+      new Function<CharSequence, String>() {
+      public String apply(CharSequence seq) {
+        return seq.toString();
+      }
+    };
+
+  public bigQueryInterpreter(Properties property) {
+    super(property);
+  }
+
+
+  //Function to return valid BigQuery Service
+  @Override
+  public void open() {
+    if (service == null) {
+      synchronized (serviceLock) {
+        if (service == null) {
+          try {
+            service = createAuthorizedClient();
+            exceptionOnConnect = null;
+            logger.info("Opened BigQuery SQL Connection");
+          }  catch (IOException e) {
+            logger.error("Cannot open connection", e);
+            exceptionOnConnect = e;   
+            close();
+          }
+        }
+      }
+    }
+  }
+
+  //Function that Creates an authorized client to Google Bigquery.
+  private static Bigquery createAuthorizedClient() throws IOException {
+    HttpTransport transport = new NetHttpTransport();
+    JsonFactory jsonFactory = new JacksonFactory();
+    GoogleCredential credential =  GoogleCredential.getApplicationDefault(transport, jsonFactory);
+
+    if (credential.createScopedRequired()) {
+      Collection<String> bigqueryScopes = BigqueryScopes.all();
+      credential = credential.createScoped(bigqueryScopes);
+    }
+
+    return new Bigquery.Builder(transport, jsonFactory, credential)
+        .setApplicationName("Zeppelin/1.0 (GPN:Apache Zeppelin;)").build();
+  }
+
+  //Function that generates and returns the schema and the rows as string
+  public static String printRows(final GetQueryResultsResponse response) {
+    StringBuilder msg = null;
+    msg = new StringBuilder();
+    try {
+      for (TableFieldSchema schem: response.getSchema().getFields()) {
+        msg.append(schem.getName());
+        msg.append(TAB);
+      }      
+      msg.append(NEWLINE);
+      for (TableRow row : response.getRows()) {
+        for (TableCell field : row.getF()) {
+          msg.append(field.getV().toString());
+          msg.append(TAB);
+        }
+        msg.append(NEWLINE);
+      }
+      return msg.toString();
+    }
+    catch ( NullPointerException ex ) {
+      throw new NullPointerException("SQL Execution returned an error!");
```

Comment:
Allthough it's not obvious what is the error condition here - it's alwasy better to keep the cause:
```java
throw new NullPointerException("SQL Execution returned an error!", ex);
```